c++/src/Writer.cc

```diff
@@ -542,6 +568,7 @@ namespace orc {
     // update file statistics
     std::vector<proto::ColumnStatistics> colStats;
     columnWriter->getFileStatistics(colStats);
+    fileFooter.clear_statistics();
```

Comment:
I'm not sure if this is the right place to do this. It seems to give desired result when doing the intermediate footer writes that cause stripe flush/write, but I'm thinking that it might not be OK when stripes are written as part of normal flushing due to their size.

Reply:
This should be OK, as we have anyway to overwrite file-level stats.